### PR TITLE
Updates System.Reflection.Metadata.sln to VS2015 and the project language to C# 6

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera @weshaggard Any reason not to do this?

**Comment by mellinoe** [comment]
Solution/tools version changes should be fine, but I don't think we're ready to switch to C#6 yet.

**Comment by tmat** [comment]
Why?

**Comment by tmat** [comment]
I'm moving some code from Roslyn to System.Reflection.Metadata and would really like to not waste my time by converting C# 6 syntax to C# 5.

**Comment by mellinoe** [comment]
@weshaggard Do we know when we will be able to adopt C#6?

**Comment by weshaggard** [comment]
I'm hoping soon (maybe weeks) we will be restoring the roslyn toolset package as part of our build process at which point we can open the door for C# 6.0. Right now there are still a few places which build this code using the native C# compiler which obviously doesn't support C# 6.0.

**Comment by tmat** [comment]
Are those places specifically building System.Reflection.Metadata?

File: src/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <CLSCompliant>false</CLSCompliant>
     <AssemblyVersion>1.2.0.0</AssemblyVersion>
     <PackageTargetFramework>dotnet5.2</PackageTargetFramework>
+    <LangVersion>6</LangVersion>
```
Comment (weshaggard) [inline src/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj]: We should definitely not do this for only this project. We have to wait until all of them can switch.

**Comment by weshaggard** [comment]
Yes there are places that build all the libraries including System.Reflection.Metadata. My suggestion is to wait a until the entire repo supports C# 6.0 or do the conversion.

**Comment by tmat** [comment]
We agreed to use this PR as a test too flush out all issues that might occur during upgrade to C# 6. Merging.

**Pull request merged**